### ISSUE TITLE
Do you have samples/were you able to replicate the original paper's results?

### ISSUE THREAD

**Issue by aduriseti** [pritishyuvraj/Voice-Conversion-GAN#1]
Hey - I noticed some pretty significant architecture changes from the paper (i.e. no final activations on either the discriminator or the generator)

I know you made the change to the discriminator intentionally - but was unsure about the change to the generator.

Were you able to get good  results w/ this repo?  How long did you have to train?

Best,
Amal

**Comment by aduriseti** [comment]
oh wait my bad - the original paper didn't have tanh output for the generator either - I was adding that in from the ganhacks repo

the rest of my questions stand

**Comment by aduriseti** [comment]
yeah - ok looks like you just did a wgan loss function - ok makes sense now

**Comment by pritishyuvraj** [comment]
We tried a lot to reproduce the results but we couldn't. On further investigation, we tried to implement the same model using Tensorflow and the model was converging far better. Also, we got good audio conversion in Tensorflow. 

Our conclusion was that Tensorflow better performs sparse calculations. Please check out my post on PyTorch Forum: https://discuss.pytorch.org/t/poor-convergence-on-pytorch-compared-to-tensorflow-using-adam-optimizer/31425

Please let me know if you are able to reproduce the results.

**Comment by pritishyuvraj** [comment]
We trained for 400,000+ iterations (2600+ epochs). Tensorflow gave us good audio conversion within these whereas PyTorch didn't give satisfactory conversion output.

**Comment by aduriseti** [comment]
Thanks for getting back to me - 

Did you check parameter counts b/w the pytorch and tensorflow implementations?  

Was this w/ the model.py or the model_GLU.py?  I noticed that in model.py the residual layer in the generator was not cloned 6 times but rather unrolled 6 times .

Any other ideas on the cause of the difference in performance?

**Comment by aduriseti** [comment]
Saw your post - results are quite startling - 

I'm using he init - but i don't have an identical tensorflow initialization to benchmark against

**Comment by aduriseti** [comment]
Also - I saw in your code that you tried out gmm based conversion as a baseline - how did that go?

**Comment by pritishyuvraj** [comment]
> Thanks for getting back to me -
> 
> Did you check parameter counts b/w the pytorch and tensorflow implementations?
> 
> Was this w/ the model.py or the model_GLU.py? I noticed that in model.py the residual layer in the generator was not cloned 6 times but rather unrolled 6 times .
> 
> Any other ideas on the cause of the difference in performance?

**Comment by pritishyuvraj** [comment]
Q.1) Did you check parameter counts b/w the pytorch and tensorflow implementations?
 Yes, in both models I had exact number of parameters in pytorch and tensorlfow (some 58 Million for Generator and 32Million for Discriminator)

Q.2)Was this w/ the model.py or the model_GLU.py?
The final one is with model_GLU.py (check the import statement in the code), I was trying everything out and hence model.py

Q.3) gmm based conversion as a baseline - how did that go?
Amazing!! Mind Blowing!!! GMM outperforms all of these. The only catch is that GMM expects time-aligned and parallel data for training. It converges in a matter of minutes (approx. 20 mins). Check this tutorial: https://r9y9.github.io/nnmnkwii/stable/nnmnkwii_gallery/notebooks/vc/01-GMM%20voice%20conversion%20(en).html

**Comment by aduriseti** [comment]
Great - thanks for the heads up on the gmm - for my application paired data might be feasible.

Did you ever try a 3rd dl library (e.g. chainer)?

Also - did you ever try just training the generator on paired data - also as a baseline?  I'm guessing you would have to align the data then also.

P.s. - thanks for helping me out and answering questions - youre saving hours of time and days of training - lmk if I can return the favor somehow

**Comment by pritishyuvraj** [comment]
I haven't tried any 3rd dl library.

As per my understanding of GAN's, irrespective of paired or unpaired data, they would take a good amount of time. The problem is learning the process. 

I am glad my answers saved your time. Just keep helping everyone in the community to grow probably nothing to return any favour :)
